sheep_cli/src/main.rs

```diff
@@ -208,11 +211,37 @@ fn load_images(input: &[String]) -> Vec<InputSprite> {
         .collect()
 }
 
-fn write_files<S: Serialize>(output_path: &str, outbuf: RgbaImage, meta: S, pretty: bool) {
-    outbuf
-        .save(format!("{}.png", output_path))
-        .expect("Failed to save image");
+fn write_image(output_path: &str, sheet: &SpriteSheet, compress: bool) {
+    let filename = format!("{}.png", output_path);
+
+    if compress {
+        let mut png_bytes = Vec::<u8>::new();
+        {
+            let mut encoder =
+                png::Encoder::new(&mut png_bytes, sheet.dimensions.0, sheet.dimensions.1);
+            encoder.set_color(png::ColorType::RGBA);
+            encoder.set_depth(png::BitDepth::Eight);
+            encoder.set_compression(png::Compression::Fast);
+            encoder.set_filter(png::FilterType::NoFilter);
+            let mut writer = encoder.write_header().unwrap();
```

Comment:
Please use `expect` here instead of `unwrap`.
`encoder.write_header().expect("Failed to write png header")`

Reply:
Oh, ok, I just thought it's okay cuz it can't fail cuz we are writing into a vector, not a file